service/matching/matching_engine.go

```diff
@@ -2021,6 +2021,44 @@ func (e *matchingEngineImpl) SyncDeploymentUserData(
 					delete(tqWorkerDeploymentData.Versions, buildID)
```

Comment:
need to remove from old format too.

Also, when all versions of the deployment are removed, clean up the whole deployment data from the upper map.